etna/search/views.py

```diff
@@ -1,172 +1,211 @@
-            "bucket_count_response": bucket_count_response,
-            "buckets": CATALOGUE_BUCKETS,
-        },
-    )
-
-
-def catalogue_search(request):
-
-    per_page = int(request.GET.get("per_page", 20))
-    page_number = int(request.GET.get("page", 1))
-    offset = (page_number - 1) * per_page
+    # The source data for get_buckets()
+    bucket_list: BucketList = None
+    # Can be updated by the view for get_current_bucket_key() to pick up
+    current_bucket_key: str = None
+
+    def extract_group_buckets(
+        self, api_result: Union[None, Dict[str, Any]]
+    ) -> Sequence[Dict[str, Union[str, int]]]:
+        """
+        Attempts to find and return a list or values from `api_result`
+        that can be passed to `get_buckets()`, allowing it to set the
+        `result_count` value for each bucket.
+        """
+        if not api_result:
+            return ()
+
+        # Account for different API response structures
+        aggregations = {}
+        if api_result.get("responses"):
+            aggregations = api_result["responses"][0].get("aggregations", {})
+        else:
+            aggregations = api_result.get("aggregations", {})
+
+        # The API response has an unfamiliar structure
+        if not aggregations:
+            return ()
+
+        return aggregations.get("group", {}).get("buckets", ())
+
+    def get_current_bucket_key(self):
+        return self.current_bucket_key
+
+    def get_buckets(
+        self,
+        group_buckets: List[Dict[str, Union[str, int]]] = None,
+        current_bucket_key: str = None,
+    ) -> Optional[BucketList]:
+        """
+        Returns a modified `BucketList` value representing the 'buckets'
+        that are available for the user to explore.
+
+        If `group_buckets` is provided, the data will be used to set the `result_count`
+        attribute for each bucket.
+
+        The `current_bucket_key` is provided, any bucket with a `key` value matching
+        the provided value will have it's `is_current` value set to `True`.
+        """
+        if not self.bucket_list:
+            return None
+
+        bucket_list = copy.deepcopy(self.bucket_list)
+
+        if group_buckets:
+            # set `result_count` for each bucket
+            doc_counts_by_key = {
+                group["key"]: group["doc_count"] for group in group_buckets
+            }
+            for bucket in bucket_list:
+                bucket.result_count = doc_counts_by_key.get(bucket.key, 0)
+
+        if current_bucket_key:
+            # set 'is_current=True' for the relevant bucket
+            for bucket in bucket_list:
+                if bucket.key == current_bucket_key:
+                    bucket.is_current = True
+                    break
+
+        return bucket_list
+
+    def get_context_data(self, **kwargs):
+        if self.bucket_list:
+            group_buckets = self.extract_group_buckets(
+                getattr(self, "api_request", None)
+            )
```

Comment:
I guess typo, xhanging from `api_request ` to `api_result` should show Bucket counts.

Reply:
Thanks for the diagnosis @TNA-Allan! This should be fixed in 499e04c